### PR TITLE
Add timing metrics for compaction

### PR DESCRIPTION
**Pull request by salvacorts**
**What this PR does / why we need it**:
This PR modifies the metrics of the bloom compactor to:
- Add new histogram metrics to track the time spent to process compaction cycles, tenants and jobs.
- Replace `_succeeded` and `_failed` metrics with a new `_completed` metric with a status label. 

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Review by owen-d — APPROVED** [review]

File: pkg/bloomcompactor/bloomcompactor.go [hunk 1]
```diff
@@ -211,13 +211,16 @@ func (c *Compactor) running(ctx context.Context) error {
 	for {
 		select {
 		case <-ticker.C:
+			start := time.Now()
```
Comment (chaudum) [inline pkg/bloomcompactor/bloomcompactor.go:214]: convenience: `<- ticker.C` yields the time of the tick

File: pkg/bloomcompactor/bloomcompactor.go [hunk 2]
```diff
@@ -211,13 +211,16 @@ func (c *Compactor) running(ctx context.Context) error {
...
 			c.metrics.compactionRunsStarted.Inc()
 			if err := c.runCompaction(ctx); err != nil {
-				c.metrics.compactionRunsFailed.Inc()
+				c.metrics.compactionRunsCompleted.WithLabelValues("fail").Inc()
```
Comment (chaudum) [inline pkg/bloomcompactor/bloomcompactor.go:217]: Do we have common name for the label value in the failure case? I think we should try to be consistent with existing label names.

**Comment by chaudum** [comment]
The e2e test rely on one of these metrics, see https://github.com/grafana/loki/blob/main/integration/loki_micro_services_test.go#L1223-L1230 I think you will need to adjust the assertion there.

File: pkg/bloomcompactor/bloomcompactor.go (hunk 2 above)
Comment (salvacorts) [reply pkg/bloomcompactor/bloomcompactor.go:217]: Good call. I see we use "failure" for failed statuses. I've added constants for both statuses.

**Review by chaudum — APPROVED** [review]

**Comment by salvacorts** [comment]
The `TestBloomFiltersEndToEnd` started to be flanky today since we set the client time to `Fri Jan 19 2024 00:00` but the server's now is the actual _now_ date. Since the max_table_age is 7 days, the test started to fail today.

This PR fixes it as well, but the `test_target_branch` drone stage will fail until we merge this PR.

**Pull request merged**